pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -184,6 +184,7 @@ let
 
       # needed for ss
       INET_DIAG         = yes;
+      INET_TCP_DIAG     = yes;
```

Comment:
Detail that seems unimportant: in current master we have `module` but "you" switched it to `yes`... that's an intentional difference from the following two?

Reply:
Yes, it was intentional. `module` doesn't work for `INET_TCP_DIAG` (on the 5.4 kernel, at least):

```
error: option not set correctly: INET_TCP_DIAG (wanted 'm', got 'y')
```

This is the relevant kernel code:
```
config INET_TCP_DIAG
        depends on INET_DIAG
        def_tristate INET_DIAG
```

I'm guessing this happens because when `INET_DIAG` is `yes` then `INET_TCP_DIAG` can't be a module. That said, I don't know exactly where is (or why there is) this restriction, as the `Kconfig` language documentation doesn't seem to say that the `depends on` or `def_tristate` definitions cause this restriction.

Reply:
We should make `INET_DIAG` a module than.

Reply:
Done, thanks!